internal/configs/config_test.go

```diff
@@ -195,7 +224,7 @@ func TestConfigProviderRequirementsInclTests(t *testing.T) {
 	terraformProvider := addrs.NewBuiltInProvider("terraform")
 	configuredProvider := addrs.NewDefaultProvider("configured")
 
-	got, diags := cfg.ProviderRequirements()
+	got, qualifs, diags := cfg.ProviderRequirements() // TODO: Question: should this handle also the test providers?
```

Comment:
I don't think we need to gather providers defined in test files. If we have an invalid configuration (from the perspective of this particular issue), then test files don't matter much, otherwise - they must not be able to "break" provider references.

Reply:
I think [Martin's comment](https://github.com/opentofu/opentofu/pull/2479#discussion_r1945255473) is already tackling that.
I will rework this while working on his comment.